src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs

```diff
@@ -492,5 +492,75 @@ private static string GetSimpleTypeName(SimpleNameSyntax name)
         private bool IsExtensionMethod(MethodDeclarationSyntax method)
             => method.ParameterList.Parameters.Count > 0 &&
                method.ParameterList.Parameters[0].Modifiers.Any(SyntaxKind.ThisKeyword);
+
+        public override string GetRootNamespace(CompilationOptions compilationOptions)
+            => string.Empty;
+
+        public override bool TryGetTargetTypeName(SyntaxNode node, out string targetTypeName)
+        {
+            if (node is MethodDeclarationSyntax methodDeclaration && IsExtensionMethod(methodDeclaration))
+            {
+                targetTypeName = GetTypeName(methodDeclaration.ParameterList.Parameters[0].Type);
+                return true;
```

Comment:
this is strane.  GetTypeName can return null, but you return true here in that case.  that seems wrong.

Reply:
Added some comment here. Basically, `targetTypeName` = null means this is a "complex method"